covsirphy/cleaning/jhu_data.py

```diff
@@ -546,6 +547,9 @@ def subset_complement(self, country, province=None,
             df, interval=interval, max_ignored=max_ignored)
         # Whether complemented or not
         is_complemented = not df.equals(subset_df)
+        # Apply absolute value to recovered records and sort them in ascending order
+        df[self.R] = df[self.R].abs()
```

Comment:
Why these codes were added outside of `._complement_recovered_full()`?

If necessary, please update `is_complemented` when these codes change recovered data.

Reply:
Why `is_complemented` would need to be updated? It is updated once at the end of these complementary methods, at the position it is already located.
I added these two lines in `subset_complement()` as well, for the same reason the `df = df.loc[df[self.R] > 0, :]` exists. Because I thought that, apart from `_complement_recovered_full()`, R could be set to negative from elsewhere. Maybe you meant with this line to keep all the non-zero only, not the negative specifically.
I added this as a last safety way to ensure positiveness and ascending order. But if you think that these R negatives come exclusively from the `_complement_recovered_full()` method, then we are okay to remove these lines.

Reply:
OK, sorting will be done after `_complement_recovered_full` and `_complement_recovered_partial`.


Sorting is a kind of complement, but `df = df.loc[df[self.R] > 0, :]` is not a complement because the former is not done when `.subset()` and the latter is also done in `.subset()`. Sorting will be done only when `.subset_complement()`.

This is why we need the following order.
```Python
df = self._complement_recovered_partial( df, interval=interval, max_ignored=max_ignored)
df[self.R] = sorted(df[self.R].abs())
is_complemented = not df.equals(subset_df) 
df = df.loc[df[self.R] > 0, :]
```

Reply:
`df = df.loc[df[self.R] > 0, :]` drops 0 and negative values, without changing values.

Reply:
Okay so I just need to put this before
```
is_complemented = not df.equals(subset_df) 
df = df.loc[df[self.R] > 0, :]
```

Reply:
But we keep the sorting inside `_complement_recovered_full` as well, right?

Reply:
Yes because sorting is required before checking the variables show monotonic increasing with `JHUData.._complement_non_monotonic()`.

Reply:
But `_complement_non_monotonic()` is called right before `_complement_recovered_full`, how do you mean it?

Reply:
Yes I agree, since `_complement_non_monotonic()` is called inside `_complement_recovered_full` as well.